packages/tools/src/tools/AdvancedMagnifyTool.ts

```diff
@@ -0,0 +1,718 @@
+import { AnnotationTool } from './base';
+
+import {
+  getEnabledElement,
+  eventTarget,
+  triggerEvent,
+  utilities as csUtils,
+} from '@cornerstonejs/core';
+import type { Types } from '@cornerstonejs/core';
+
+import {
+  addAnnotation,
+  getAnnotations,
+} from '../stateManagement/annotation/annotationState';
+import { isAnnotationLocked } from '../stateManagement/annotation/annotationLocking';
+import { isAnnotationVisible } from '../stateManagement/annotation/annotationVisibility';
+import {
+  drawCircle as drawCircleSvg,
+  drawHandles as drawHandlesSvg,
+} from '../drawingSvg';
+import { state } from '../store';
+import { Events, MouseBindings, KeyboardBindings } from '../enums';
+import { getViewportIdsWithToolToRender } from '../utilities/viewportFilters';
+import {
+  resetElementCursor,
+  hideElementCursor,
+} from '../cursors/elementCursor';
+import {
+  EventTypes,
+  ToolHandle,
+  PublicToolProps,
+  ToolProps,
+  SVGDrawingHelper,
+} from '../types';
+import { AdvancedMagnifyAnnotation } from '../types/ToolSpecificAnnotationTypes';
+
+import { AnnotationCompletedEventDetail } from '../types/EventTypes';
+import triggerAnnotationRenderForViewportIds from '../utilities/triggerAnnotationRenderForViewportIds';
+import { StyleSpecifier } from '../types/AnnotationStyle';
+import { getCanvasCircleRadius } from '../utilities/math/circle';
+import AdvancedMagnifyViewportManager from './AdvancedMagnifyViewportManager';
+import type { AutoPanCallbackData } from './AdvancedMagnifyViewport';
+
+class AdvancedMagnifyTool extends AnnotationTool {
+  static toolName;
+
+  magnifyViewportManager: AdvancedMagnifyViewportManager;
+  touchDragCallback: any;
+  mouseDragCallback: any;
+  editData: {
+    annotation: any;
+    viewportIdsToRender: Array<string>;
+    handleIndex?: number;
+    newAnnotation?: boolean;
+    hasMoved?: boolean;
+  } | null;
+  isDrawing: boolean;
+
+  constructor(
+    toolProps: PublicToolProps = {},
+    defaultToolProps: ToolProps = {
+      supportedInteractionTypes: ['Mouse', 'Touch'],
+      configuration: {
+        shadow: true,
+        magnifyingGlass: {
+          radius: 125, // px
+          zoomFactor: 2.5,
+          zoomFactorList: [2.5, 3, 3.5, 4, 4.5, 5],
+          autoPan: {
+            enabled: true,
+            padding: 10, // px
+          },
+        },
+        actions: [
+          {
+            method: 'showZoomFactorsList',
+            bindings: [
+              {
+                mouseButton: MouseBindings.Secondary,
+                modifierKey: KeyboardBindings.Shift,
+              },
+            ],
+          },
+        ],
+      },
+    }
+  ) {
+    super(toolProps, defaultToolProps);
+    this.magnifyViewportManager = AdvancedMagnifyViewportManager.getInstance();
+  }
+
+  private _getZoomFactorsListDropdown(currentZoomFactor, onChangeCallback) {
```

Comment:
re-order so that addNewAnnotation is first similar to others, also publics first then private fns